.github/workflows/main.yml

```diff
@@ -88,17 +86,65 @@ jobs:
           docker-compose -f ./docker-compose.test.yml up -d
           docker-compose -f ./docker-compose.test.yml exec -T river pytest -vv -m "not django_db"
 
+  test-hapi:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: '15'
+      # runs the Maven `package` target in non-interactive mode to ensure
+      # that the code builds, tests pass, and a package can be created.
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots verify
+        working-directory: hapi-loader
+
+  build-and-publish-hapi:
+    needs: test-hapi
```

Comment:
are we sure we want the tests to pass before building/publishing hapi loader images ? it's not the case for other river images and it would speed up the build if we didn't wait.
also, can be useful when we want to test on dev but we didn't fix unit tests yet. what's your opinion ?

Reply:
Fine by me
For the other images, we publish beforehand because we perform the tests inside the docker image

Reply:
yes I remember but I'm always up for shorter build times ;)